.github/workflows/generate-and-deploy-with-delete.yml

```diff
@@ -2,31 +2,60 @@
 name: Generate and Sync Files with Delete
 
 on:
-  workflow_dispatch
+  workflow_dispatch:
+    inputs:
+      environment:
+        type: choice
+        description: The environment of the Registry
+        options:
+          - Development
+          - Production
+      bump-versions-branch:
+        type: string
+        description: The branch with the modules and providers to publish (if not specified will use the workflow's ref)
+        required: false
 
 jobs:
   generate-and-sync:
     runs-on: ubuntu-latest
-
-    environment: Production-with-delete
+    environment:
+      name: ${{ inputs.environment }}
 
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v4
+        with:
+          path: main
+
+      - name: Checkout branch with modules and providers
+        if: inputs.bump-versions-branch != ''
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.bump-versions-branch }}
+          path: providers-and-modules
 
       - name: Set up Go
         uses: actions/setup-go@v4
         with:
-          go-version-file: './src/go.mod'
+          go-version-file: './main/src/go.mod'
 
       - name: Setup dependencies
         run: sudo apt-get install rclone
 
       - name: Run Generation script
-        working-directory: ./src
-        run: go run ./cmd/generate-v1 --destination ../generated
+        working-directory: ./main/src
+        run: |
+          PATH_PREFIX="../../providers-and-modules"
+          
+          if [ -z "${{ inputs.bump-versions-branch }}" ]; then
+            DATA_DIR_FLAGS=""
+          else
+            DATA_DIR_FLAGS="--module-data $PATH_PREFIX/modules --provider-data $PATH_PREFIX/providers --key-data $PATH_PREFIX/keys"
+          fi
```

Comment:
Why is this conditional necessary? I don't see a reason those flags should be different if this is running on a custom branch or not

Reply:
If I understand correctly, this is using the workflow branch for the ./repo directory and optionally allowing to override the providers/modules specified. via   I'm not sure when this is necessary?

Reply:
If we don't have a custom branch the source of our modules and providers is different. In this case, we'll use the providers and modules that exist on the WF's branch (`../providers` which is the default in the script if the flag wasn't provided).

Reply:
Left a suggestion for not having to use a custom branch or `./repo` directory

Reply:
I took your suggestion 🙃